### DISCUSSION TITLE
Fit control width to content (for singleValue)

### DISCUSSION THREAD

**Discussion by clmz** [JedWatson/react-select#5139] [Q&A]
Hello,

I know there's a possibility to fit the menu to its longest option via 
`width: 'max-content, minWidth: '100%`

I cannot find a way to apply this behaviour to the control component, so it can act e.g. like a button and fit automaticcally the width to the value / selected olption. 
The problem is the absolute position of the singleValue - changing this has an impact to the cursor position while searching as well as the width of the field.

Any ideas?

**Comment by sinchang** [comment]
Hi @clmz, Do you figure out the solution?

**Reply by sinchang** [reply]
You can check out the [example](https://codesandbox.io/s/react-select-longest-option-width-geunu?file=/src/App.js) created by react-select Collaborator. 

Hope it helps you.
